onnxruntime/contrib_ops/cuda/bert/add_bias_transpose.cu

```diff
@@ -651,11 +651,14 @@ void InvokeAddBiasTranspose(
     if (format != 1 && format != 2 && format != 3) {
       ORT_THROW("format must be 1, 2 or 3 for rotary attention");
     }
+    if (qk_head_size != 64 && qk_head_size !=128) {
```

Comment:
nit: if add bias transpose does not have assumption of head size, we shall remove it here. I saw you have similar check in decoder_masked_self_attention.cc.

Reply:
the rotary embedding in add_bias_transpose() assumes these, but it might be better to move it to attention_base.cc.
Will do in future PRs. Currently it prevents silent bugs from calling the function from other ops like MHA.